react/components/ProgressBar/utils.ts

```diff
@@ -35,11 +36,18 @@ export function generateProgressBarStates(
   })
 }
 
+export function getCurrentProgressBarState(status: string, packages: any) {
+  const currentProgressIndex = getOrderProgress(status, packages)
+
+  return generateProgressBarStates(progressBarStates, currentProgressIndex, packages)[currentProgressIndex]?.label
+}
+
 export function isDelivered(packages: any) {
-  let isDelivered = true
-  if (packages.length === 0) {
-    isDelivered = false
+  if (packages == null || packages.length === 0) {
+    return false
   }
+
+  let isDelivered = true
```

Comment:
You could use `Array.prototype.some` here instead of a control variable.
But anyway, not the point of this PR.

Reply:
Agreed! But I won't change it now to avoid creating more variables that could result in bugs